Source/KSCrash/Source/KSCrash/Recording/BSG_KSCrashC.c

```diff
@@ -63,11 +63,51 @@ static char *bsg_g_stateFilePath;
 // ============================================================================
 #pragma mark - Utility -
 // ============================================================================
+static const int bsg_filepath_len = 512;
+static const int bsg_error_class_filepath_len = 21;
 
 static inline BSG_KSCrash_Context *crashContext(void) {
     return &bsg_g_crashReportContext;
 }
 
+int bsg_create_filepath(char *base, char filepath[bsg_filepath_len], char severity, char error_class[bsg_error_class_filepath_len]) {
+    int length;
+    for (length = 0; length < bsg_filepath_len; length++) {
+        if (base[length] == '\0') {
+            break;
+        }
+        filepath[length] = base[length];
+    }
+    if (length > 5) // Remove initial .json from path
+        length -= 5;
+
+    // append contextual info
+    BSG_KSCrash_Context *context = crashContext();
+    filepath[length++] = '-';
+    filepath[length++] = severity;
+    filepath[length++] = '-';
+    filepath[length++] = context->crash.crashType == BSG_KSCrashTypeUserReported ? 'h' : 'u';
```

Comment:
AppCode warns that this is attempting to fit an `int` into a `char`, we could add an explicit cast to suppress this.

Reply:
Not sure why it would feel that way, considering that `'h'` and `'u'` are chars.